packages/hoppscotch-common/src/services/oauth/utils.ts

```diff
@@ -108,3 +108,15 @@ export const refreshToken = async ({
     ? E.right(parsedTokenResponse.data)
     : E.left("AUTH_TOKEN_REQUEST_INVALID_RESPONSE" as const)
 }
+
+/**
+ * Common OAuth2 parameter schema with all possible fields
+ * Used as base for both auth requests and advanced token/refresh requests
+ */
+export const OAuth2ParamSchema = z.object({
+  id: z.number(),
+  key: z.string(),
+  value: z.string(),
+  active: z.boolean(),
+  sendIn: z.enum(["headers", "url", "body"]).optional(),
+})
```

Comment:
We'll have to revisit the duplicated type definitions.

https://github.com/hoppscotch/hoppscotch/blob/c04faaaf27e756f8e507942e6ab771503b3b91d1/packages/hoppscotch-data/src/rest/v/15/auth.ts#L22-L29

Reply:
Tracked in [FE-1004](https://linear.app/hoppscotch/issue/FE-1004/streamline-duplicated-advanced-param-type-definitions-associated-with).